scripts/storage-emulator-integration/internal/tests.ts

```diff
@@ -0,0 +1,168 @@
+import { expect } from "chai";
+import * as supertest from "supertest";
+import { StorageRulesFiles } from "../../../src/test/emulators/fixtures";
+import { TriggerEndToEndTest } from "../../integration-helpers/framework";
+import {
+  EMULATORS_SHUTDOWN_DELAY_MS,
+  getStorageEmulatorHost,
+  readEmulatorConfig,
+  TEST_SETUP_TIMEOUT,
+} from "../utils";
+
+const FIREBASE_PROJECT = process.env.FBTOOLS_TARGET_PROJECT || "fake-project-id";
+const EMULATOR_CONFIG = readEmulatorConfig();
+const STORAGE_EMULATOR_HOST = getStorageEmulatorHost(EMULATOR_CONFIG);
+
+describe("Storage emulator internal endpoints", () => {
+  let test: TriggerEndToEndTest;
+
+  before(async function (this) {
+    this.timeout(TEST_SETUP_TIMEOUT);
+    process.env.STORAGE_EMULATOR_HOST = STORAGE_EMULATOR_HOST;
+    test = new TriggerEndToEndTest(FIREBASE_PROJECT, __dirname, EMULATOR_CONFIG);
+    await test.startEmulators(["--only", "auth,storage"]);
+  });
+
+  beforeEach(async () => {
+    // Reset emulator to default rules.
+    await supertest(STORAGE_EMULATOR_HOST)
+      .put("/internal/setRules")
+      .send({
+        rules: {
+          files: [StorageRulesFiles.readWriteIfAuth],
+        },
+      })
+      .expect(200);
```

Comment:
In one of the tests can we read the rules back and assert they were set to the expected value?